### PR TITLE
Adding Colour Selection to add and remove flowers

### PR DESCRIPTION
**Pull request by samuelory**
As some flowers have different meanings based on what colour they are, it is necessary to make sure the colour of flower the user is adding to their bouquet is the correct one.

### CODE REVIEW

**Comment by samuelory** [comment]
Also I edited the runtest to include more commands, so had to trim a lot of the messages printed to the console

File: src/main/java/florizz/core/Ui.java [hunk 1]
```diff
@@ -393,4 +394,56 @@ public void printSaveSuccess(String bouquetName) {
...
+            try {
+                switch (colourInput) {
+                case "back":
+                    printBackPage();
+                    break;
+                case "next":
+                    printNextPage();
+                    break;
+                case "cancel":
+                    return new Flower();
```
Comment (Ijaaz01) [inline src/main/java/florizz/core/Ui.java:418]: Cool idea to implement

File: src/main/java/florizz/storage/Storage.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public void saveAllBouquets(FileWriter bouquetStorageWriter, ArrayList<Bouquet>
             for (Map.Entry<Flower, Integer> k : tempHashMap.entrySet()) {
                 Flower flower = k.getKey();
                 Integer quantity = k.getValue();
-                bouquetStorageWriter.write("add " + flower.getFlowerName() +
+                bouquetStorageWriter.write("add " + flower.getFlowerName() + " /c " + flower.getColour() +
```
Comment (Ijaaz01) [inline src/main/java/florizz/storage/Storage.java:73]: Good catch regarding the automatic storage

File: src/main/java/florizz/core/Parser.java [hunk 3]
```diff
@@ -261,7 +287,18 @@ private static RemoveFlowerCommand handleRemoveFlower(String argument) throws Fl
...
+            try{
+                flowerName = argument.substring(0,colourIndex).trim();
+                String colourString = removePrefix(argument.substring(colourIndex, quantityIndex), COLOUR).trim();
+                Flower.Colour colourToAdd = Flower.stringToColour(colourString);
+                return new RemoveFlowerCommand(flowerName, colourToAdd, quantity, bouquetName);
+            } catch(IllegalArgumentException error){
+                throw new FlorizzException("Tried to add a non recognised colour" +
+                        "Type 'flowers' to view all the currently available flowers and their colours.");
+            }
+        }
```
Comment (Ijaaz01) [inline src/main/java/florizz/core/Parser.java:301]: Good way to handle the colour being option

**Review by Ijaaz01 — APPROVED** [review]
Great work cant wait to try out the features

**Pull request merged**